ludwig/schema/features/augmentation/utils.py

```diff
@@ -0,0 +1,156 @@
+from dataclasses import field
+from typing import List, Union
+
+from marshmallow import fields, ValidationError
+
+from ludwig.api_annotations import DeveloperAPI
+from ludwig.constants import TYPE
+from ludwig.schema import utils as schema_utils
+from ludwig.utils.registry import Registry
+
+_augmentation_config_registry = Registry()
+
+
+@DeveloperAPI
+def get_augmentation_config_registry() -> Registry:
+    return _augmentation_config_registry
+
+
+@DeveloperAPI
+def register_augmentation_config(name: str, features: Union[str, List[str]]):
+    if isinstance(features, str):
+        features = [features]
+
+    def wrap(cls):
+        for feature in features:
+            augmentation_registry = get_augmentation_config_registry().get(feature, {})
+            augmentation_registry[name] = cls
+            get_augmentation_config_registry()[feature] = augmentation_registry
+        return cls
+
+    return wrap
+
+
+@DeveloperAPI
+def get_augmentation_cls(feature: str, name: str):
+    return get_augmentation_config_registry()[feature][name]
+
+
+@DeveloperAPI
+def get_augmentation_classes(feature: str):
+    return get_augmentation_config_registry()[feature]
+
+
+@DeveloperAPI
+def AugmentationContainerDataclassField(feature_type: str, default=[], description=""):
```

Comment:
nit: let's be super explicit this and call this <pre>Augmentation<i>**List**</i>DataclassField</pre>. If this class actually handled the boolean case for augmentations then "container" would be appropriate (since it would be handling everything), but this class _only_ handles _lists_.

Reply:
done